qiskit/transpiler/__init__.py

```diff
@@ -155,6 +229,588 @@
+
+This routing algorithm runs with threaded parallelism to consider several different possibilities
+for routing, choosing the one that minimizes the number of inserted swaps.
+
+The optimization level affects how many different stochastic seeds are attempted for the full
+routing, and the amount of work done by :class:`.VF2PostLayout` to post-optimize the initial layout.
+
+This is almost invariably the best-performing built-in plugin, and the one Qiskit uses by default in
+all cases where routing is necessary.
+
+.. _transpiler-preset-stage-translation:
+
+Translation Stage
+-----------------
+
+.. seealso::
+    `Translation stage explanation`__
+        Higher-level user-facing explanation of the translation stage in the IBM Quantum guide.
+
+.. __: https://docs.quantum.ibm.com/guides/transpiler-stages#translation-stage
+
+The translation stage is responsible for rewriting all gates in the circuit into ones that are
+supported by the target ISA.  For example, if a ``cx`` is requested on hardware qubits 0 and 1, but
+the ISA only contains a ``cz`` operation on those qubits, the translation stage must find a way of
+representing the ``cx`` gate using the ``cz`` and available one-qubit gates.
+
+.. note::
+
+    In the Qiskit 1.x series, translation plugins need not output gates with the correct
+    directionality, provided the gate exists with opposite directionality on the given qubit pair.
+    For example, if ``cx(0, 1)`` is ISA-supported, the translation stage is permitted to output
+    ``cx(1, 0)``.
+
+    This is likely to change in later versions of Qiskit.
+
+The translation stage is called before entering the optimization stage. Optimization plugins
+(including Qiskit's built-in plugins) may also use the translation stage as a "fixup" stage after
+the optimization loop, if the optimization loop returns a circuit that includes non-ISA gates.  This
+latter situation is fairly common; the optimization loop may only be concerned with minimizing
+properties like "number of two qubit gates", and will leave its output in terms of locally
+equivalent gates, which the translation stage can easily rewrite without affecting the target
+optimization properties.  This allows easier separation of concerns between the two stages.  Some
+optimization plugins may be stricter in their output, and so this follow-up to the translation stage
+may no longer be necessary.
+
+When writing :ref:`stage plugins <transpiler-preset-stage-plugins>`, the entry point for
+``translation`` is ``qiskit.transpiler.translation``.  The built-in plugins are:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Summary
+
+    * - :ref:`translator <transpiler-preset-stage-translation-translator>`
+      - Symbolic translation of gates to the target basis using known equivalences.
+
+    * - :ref:`synthesis <transpiler-preset-stage-translation-synthesis>`
+      - Collect each run of one- and two-qubit gates into a matrix representation, and resynthesis
```

Comment:
```suggestion
      - Collect each run of one- and two-qubit gates into a matrix representation, and resynthesis
```
should it be resynthesize?

Reply:
yes